docs/cluster-agent/HORIZONTAL_POD_AUTOSCALING.md

```diff
@@ -0,0 +1,292 @@
+# Datadog Cluster Agent - DCA | Getting Started
+
+The DCA is a **beta** feature, if you are facing any issues please reach out to our [support team](http://docs.datadoghq.com/help)
+
+## Introduction
+
+The Horizontal Pod Autoscaling feature has been introduced in [Kubernetes v1.2](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/#before-you-begin).
+It allowed users to autoscale off of basic metrics like CPU, but required a resource called the metrics-server to run along side your application.
+As of Kubernetes v1.6, it is possible to autoscale off of [custom metrics](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#support-for-custom-metrics).
+Custom metrics are user defined and are collected from within the cluster.
+As of Kubernetes v1.10, the External Metrics support was introduced. Users are now able to autoscale off of any metric not restricted to the cluster.
+
+Custom and external metrics providers as opposed to the metrics server, is a resource that has to be implemented and registered by the user.
+Vendor backed implementations are starting to surface, so users can also rely on them.
+
+As of v1.0.0, the Datadog Cluster Agent (DCA) implements the External Metrics Provider interface.
+This walk through will explain how to set it up and how it can help you autoscale your Kubernetes infrastructure based off of your Datadog metrics.
+
+
+## Requirements
+
+The first requirement is to have Kubernetes v1.10, this is so the External Metrics Provider resource can be registered agains the API Server.
+Secondly, you will need to have the Aggregation layer enabled as well, please refer to the official guide:
+https://kubernetes.io/docs/tasks/access-kubernetes-api/configure-aggregation-layer/
+
+## Walk through
+
+### Security premise
+
+You can create the secret that will be used for your agents to communicate with the DCA. 
+While this is not necessary for this walk through, it is the best way to enrich the metadata of the metrics collected by your node agents with cluster level insight.
+You can modify the value in the file manifests/cluster-agent/dca-secret.yaml then create it:
+
+`kubectl create -f manifests/cluster-agent/dca-secret.yaml` 
+
+Will yield:
+
+```
+kubectl get secret datadog-auth-token
+NAME                 TYPE      DATA      AGE
+datadog-auth-token   Opaque    1         16s
+
+```
```

Comment:
I think this section is confusing in this doc. Is this already included in other DCA documentation?

Reply:
It is, and it's not necessary for the actual HPA - But I think this doc will draw more traffic than the other ones, so I want to be redundant as the secret is a major part of the DCA - Node agent communication.